data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,498 @@
+    """
+    Queries the table that retraction will take place, to see if the PID exists before creating query
+    :param pid: person_id
+    :param date_row: row that is being iterated through to create query
+    :param project_id: bq name of project_id
+    :return: Boolean if pid exists in table
+    """
+    check_pid_df = bq.query(q=CHECK_PID_EXIST_QUERY.render(
+        project=date_row.project_id,
+        dataset=date_row.dataset_id,
+        table=date_row.table,
+        pid=pid),
+                            project_id=project_id)
+    return check_pid_df.get_value(0, 'count')
+
+
+def create_queries(project_id, ticket_number, pids_project_id, pids_dataset_id,
+                   pids_table):
+    """
+    Creates sandbox and truncate queries to run for EHR deactivated retraction
+
+    :param project_id: bq name of project
+    :param ticket_number: Jira ticket number to identify and title sandbox table
+    :param pids_project_id: deactivated ehr pids table in bq's project_id
+    :param pids_dataset_id: deactivated ehr pids table in bq's dataset_id
+    :param pids_table: deactivated pids table in bq's table name
+    :return: list of queries to run
+    """
+    queries_list = []
+    dataset_list = set()
+    # Hit bq and receive df of deactivated ehr pids and deactivated date
+    deactivated_ehr_pids_df = bq.query(q=DEACTIVATED_PIDS_QUERY.render(
+        project=pids_project_id, dataset=pids_dataset_id, table=pids_table),
+                                       project_id=project_id)
+
+    date_columns_df = get_date_info_for_pids_tables(project_id)
+    LOGGER.info(
+        "Dataframe creation complete. DF to be used for creation of retraction queries."
+    )
+
+    LOGGER.info(
+        "Looping through the deactivated PIDS df to create queries based on the retractions needed per PID table"
+    )
+    for ehr_row in deactivated_ehr_pids_df.itertuples(index=False):
+        for date_row in date_columns_df.itertuples(index=False):
+            # Determine if dataset is deid to correctly pull pid or research_id and check if ID exists in dataset or if
+            # already retracted
+            if re.match(DEID_REGEX, date_row.dataset_id):
+                pid = get_research_id(date_row.project_id, date_row.dataset_id,
+                                      ehr_row.person_id)
+            else:
+                pid = ehr_row.person_id
+
+            # Check if PID is in table
+            if pid is not None and check_pid_exist(pid, date_row,
+                                                   project_id) > 0:
+                dataset_list.add(date_row.dataset_id)
+                # Get or create sandbox dataset
+                sandbox_dataset = check_and_create_sandbox_dataset(
+                    date_row.project_id, date_row.dataset_id)
+
+                # Create queries based on if the date field is null, if True, will create query based on end_date/start_date
+                if not pd.isnull(date_row.date_column):
```

Comment:
nit:  This isn't something that has to be changed.  This should work as is.  This is just a comment on the thought process.  When possible, I try to make the conditional logic statements as easy to understand as  I can.  For this one, I might have used `if pd.isnull(date_row.date_column):` then followed that with the queries to build based on end_date/start_date.  And had the `else` statement for the date field query generation.  This removes needing to negate the statement and understand the `if` logic only executes when `isnull` is `False`.

Reply:
you're right I prefer to adjust this anyway, thank you!